nextclade/nextclade.cwl

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env cwl-runner
+cwlVersion: v1.0
+class: CommandLineTool
+
+doc: Assign Nextstrain clades to SARS-CoV-2 sequences and provide QC information
+id: nextclade
+label: Nextclade
+
+dct:creator:
+  "@id": "https://orcid.org/0000-0001-6553-5274"
+  foaf:name: Peter van Heusden
+  foaf:mbox: "mailto:pvh@sanbi.ac.za"
+
+requirements:
+  DockerRequirement:
+    dockerPull: neherlab/nextclade:0.8.1-alpine
+
+hints:
+  ResourceRequirement:
+    coresMin: 1
+    ramMin: 512  # 512 MB
+
+inputs:
+  input_fasta:
+    type: File
+    doc: .fasta or .txt file with input sequences
+    format: 
+      - edam:format_1929  # FASTA
+      - edam:format_1964  # plain text format
+    inputBinding:
+      prefix: --input-fasta
+  input_qc_config:
+    type: File?
+    doc: QC config json file containing custom QC configuration
+    format: edam:format_3464  # JSON
+    inputBinding:
+      prefix: --input-qc-config
+  input_root_seq:
+    type: File?
+    doc: plain text file containing custom root sequence
+    format: edam:format_1964
+    inputBinding:
+      prefix: --input-root-seq
+  input-tree:
+    type: File?
+    doc: Auspice JSON v2 file containing custom reference tree
+    format: edam:format_3464
+    inputBinding:
+      prefix: --input-tree
+  input-gene-map:
+    type: File?
+    doc: 'JSON file containing custom gene map. Gene map (sometimes also called "gene annotations") is used to resolve aminoacid changes in genes.'
+    format: edam:format_3464
+    inputBinding:
+      prefix: --input-gene-map
+  input-pcr-primers:
+    type: File?
+    doc: CSV file containing a list of custom PCR primer sites. These are used to report mutations in these sites.
+    format: edam:format_3572
+    inputBinding:
+      prefix: --input-pcr-primers
+  output_options:
+    type:
+      type: record
+      name: output_options_record
+      fields:
+        output_json_filename:
```

Comment:
Do all of these filenames need to be specialized? Why not take the name root of the FASTA and add the relevant file extension?

Reply:
The underlying tool has options for different output names for each output file. While I am not sure if that will be used in practice, I have followed the underlying tool in the CWL here.

Reply:
Nextclade dev here. Let us know, folks, what would be the most convenient set of flags. We are flexible here.

Reply:
For "user" I meant the user of this CWL description. For nextclade itself @ivan-aksamentov you did the correct thing to ask for the filename/path.

Is there a cost to output all these types?

@pvanheus If not then I recommend always requesting all output types via the `arguments` section building off of the nameroot of the fast afile